### PR TITLE
Create secret social media platform

### PR DESCRIPTION
**Pull request by MkrtichSargsyan**
This project creates a secret social media platform where only authorized members can see who posted a ''crazy task''. This project was build as a result of inspiration by the novel of Chuck Palahniuk

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,3 +1,8 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
+  devise_for :users
+  resources :posts # , only: [:index, :show, :update, :destroy]
+  resources :users # , except: [:new]
```
Comment (acushlakoncept) [inline config/routes.rb]: Kindly update your application routes to use `resources :posts, only: [:new, :create, :index]` as this is one of the project requirements ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/posts_routes_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))

File: app/views/layouts/_alerts.html.erb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  </div>
+<% end %>
+<% if alert %>
+  <div class="alert alert-warning w-50 mx-auto">
+    <%= alert %>
+    <button type="button" class="close" data-dismiss="alert" aria-label="Close">
+      <span aria-hidden="true">&times;</span>
+    </button>
+  </div>
+<% end %>
```
Comment (acushlakoncept) [inline app/views/layouts/_alerts.html.erb]: There should be absolutely NO logic in views. >Views are only for displaying information that became available to it from its corresponding controller not generating it.

File: app/views/layouts/_navigation.html.erb [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+          <%= link_to "#{current_user.name}", edit_user_registration_path,class:"nav-link" %>
+        </li>
+        <li class="nav-item">
+          <%= link_to "logout", destroy_user_session_path,class:"nav-link", method: :delete  %>
+        </li>
+      <% else %>
+        <li class="nav-item">
+          <%= link_to 'Log in / Sign up', new_user_registration_path,  class: 'nav-link' %>
+        </li>
+      <% end %>
```
Comment (acushlakoncept) [inline app/views/layouts/_navigation.html.erb]: There should be absolutely NO logic in views. >Views are only for displaying information that became available to it from its corresponding controller not generating it.

File: app/views/posts/_post.html.erb [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+                    <%= image_tag("avatar.jpg", alt: "avatar",class:'avatar_image') %>
+                  <% end %>                                    
+                </div>
+                <h3 class='card_username'><%= post.user.name %></h3>
+              <% else %>
+                <div class='image_container'>
+                  <%= image_tag("avatar.jpg", alt: "avatar",class:'avatar_image') %>
+                </div>
+                <h3 class='card_username'>Anonymous</h3>
+              <% end %>
```
Comment (acushlakoncept) [inline app/views/posts/_post.html.erb]: There should be absolutely NO logic in views. >Views are only for displaying information that became available to it from its corresponding controller not generating it.

File: app/views/users/index.html.erb [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  <% @users.each do |user| %>
+    <div class="d-flex justify-content-md-center" >
+      <div class= 'user_card col-4 border mb-3 rounded-circle' style='background: #ad5050;'>
+        <figure class='m-0 p-3'>
+          <div class='image_container'>
+            <% if user.avatar.attached? %>
+              <%= image_tag user.avatar, alt: "avatar",class:'avatar_image' %>
+            <% else %>
+              <%= image_tag("avatar.jpg", alt: "avatar",class:'avatar_image') %>
+            <% end %>
```
Comment (acushlakoncept) [inline app/views/users/index.html.erb]: There should be absolutely NO logic in views. >Views are only for displaying information that became available to it from its corresponding controller not generating it.

File: app/models/user.rb [hunk 6]
```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
```
Comment (acushlakoncept) [inline app/models/user.rb]: Let's have a clean codebase, by removing unused/unnecessary comments. It’s better to delete commented piece of codes. If you need to review a previous version of your code, use git history ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))

File: app/views/pages/home.html.erb [hunk 7]
```diff
@@ -0,0 +1,11 @@
+<div class="home_page">
+  <%= image_tag 'theme2.jpg' %>
+  <div class="app_description d-flex justify-content-center">
+    <div style="width: 18rem;">
+      <%# <%= current_user.name %>
```
Comment (acushlakoncept) [inline app/views/pages/home.html.erb]: Let's have a clean codebase, by removing unused/unnecessary comments. It’s better to delete commented piece of codes. If you need to review a previous version of your code, use git history ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))

**Review by acushlakoncept — CHANGES_REQUESTED** [review]
# Status: Changes Required ♻️

### Hello @MkrtichSargsyan, 
Great job! 👏  I must say I am really impressed at the level you have taken this project to, super awesome. Let this good work continue. 👍  Excellent UI.

However, there are still some changes that are necessary for this project to be approved.
Let us quickly add some finishing touches and request another review, I am quite sure it won't be a problem to address the necessary changes.

- [x] Update your readme file with a missing installation instruction, `yarn install`. I needed to run that before I could `rails db:migrate`

- [x] There should be absolutely **NO logic in views**.  Views are only for displaying information that became available to it from its corresponding controller not generating it.
View-related logic can be encapsulated in View helpers (app/helpers/\*_helper.rb)

**Resources**
* [Where Do I Put My Code?](http://codefol.io/posts/Where-Do-I-Put-My-Code)
* [The Beginner's Guide to Rails Helpers](https://mixandgo.com/learn/the-beginners-guide-to-rails-helpers)
* [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/no_logic_in_helpers_requirement.png) from Pathwright

This [article ](https://medium.com/@acushla4real/rails-view-helpers-best-practices-guide-d7d7b6db3db4)would come in handy 

Kindly address the request changes above as well as the comments I have made below on your code.

### Cheers and Happy coding! 💻 👏

(Feel free to contact me on slack(@acushla), if you have any additional question)
Uduak Essien
[Twitter](https://twitter.com/acushlakoncept) | [LinkedIn](https://www.linkedin.com/in/acushlakoncept/)

**Review by kobiyoyo — APPROVED** [review]
# Project Approved :thumbsup:
## Hello Team  ,  you have done a brilliant job with this project. :clap:
#### You can merge !!!

**Pull request merged**